### PR TITLE
Fix and test HttpSys delegation

### LINKED ISSUE #28396: Add out-of-proc test for request queue delegation
All the Http.Sys tests only verify delegating requests between two different Http.Sys queues in the same process. There are some behavioral differences (such as https://github.com/dotnet/aspnetcore/pull/28336 ) that it would be good to validate

### PR DESCRIPTION
**Pull request by Tratcher**
Fixes #28396 (adding out-of-proc test)

When writing the new test we found that this feature regressed on Win11 because Http.Sys requires a new initialization flag for the feature detection API to work.

I also refactored the initialization logic so IsFeatureSupported is only called once at startup.

I plan to backport this to 6.0-rc2. We might also need to backport it to 5.0 if partners get blocked.

### CODE REVIEW

**Comment by shirhatti** [comment]
👀 @NGloreous

**Review by shirhatti — APPROVED** [review]

File: src/Servers/HttpSys/src/NativeInterop/HttpApi.cs [hunk 1]
```diff
@@ -136,18 +137,17 @@ private static void InitHttpApi(ushort majorVersion, ushort minorVersion)
             version.HttpApiMajorVersion = majorVersion;
             version.HttpApiMinorVersion = minorVersion;
 
-            var statusCode = HttpInitialize(version, (uint)HTTP_FLAGS.HTTP_INITIALIZE_SERVER, null);
+            var statusCode = HttpInitialize(version, (uint)(HTTP_FLAGS.HTTP_INITIALIZE_SERVER | HTTP_FLAGS.HTTP_INITIALIZE_CONFIG), null);
```
Comment (shirhatti) [inline src/Servers/HttpSys/src/NativeInterop/HttpApi.cs:140]: What happen on older OSes that don't have this API?

**Comment by shirhatti** [comment]
Looks like this [doc](https://docs.microsoft.com/windows/win32/api/http/nf-http-httpinitialize#parameters) needs to be updated. The functions listed that require `HTTP_INITIALIZE_CONFIG` don't include `HttpDelegateRequestEx`

File: src/Servers/HttpSys/src/NativeInterop/HttpApi.cs (hunk 1 above)
Comment (Tratcher) [reply src/Servers/HttpSys/src/NativeInterop/HttpApi.cs:140]: HTTP_INITIALIZE_CONFIG isn't a new flag, I think it's been around forever. What changed is that HttpIsFeatureSupported now requires it.

File: src/Servers/HttpSys/test/NonHelixTests/DelegateOutOfProcTests.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Http;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Testing;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Server.HttpSys.NonHelixTests
```
Comment (HaoK) [inline src/Servers/HttpSys/test/NonHelixTests/DelegateOutOfProcTests.cs:9]: Is there a reason why we don't want these to run on helix?

Comment (Tratcher) [reply src/Servers/HttpSys/test/NonHelixTests/DelegateOutOfProcTests.cs:9]: We never figured out how to make them work there. https://github.com/dotnet/aspnetcore/issues/8247

**Pull request merged**